### PR TITLE
feat(#313): implement share product popover

### PR DESCRIPTION
**Pull request by Andrew-Orlov**
Description:

--

QA-test: ST-315

Demo-test:

Download artifact URL: https://vc3prerelease.blob.core.windows.net/packages/vc-theme-b2b-vue-1.3.0-pr-192-4918d3b3.zip

### CODE REVIEW

File: client-app/shared/catalog/components/product-price-block.vue [hunk 1]
```diff
@@ -69,6 +93,8 @@ const props = defineProps({
 const { isAuthenticated } = useUser();
 const { openPopup } = usePopup();
 
+const pageUrl: string = window.location.href;
```
Comment (trueboroda) [inline client-app/shared/catalog/components/product-price-block.vue:96]: You can use type inference.

File: client-app/shared/catalog/components/product-price-block.vue [hunk 2]
```diff
@@ -30,23 +30,45 @@
...
+                class="items-center justify-center select-none py-4 px-1 border-r space-x-2 cursor-pointer hover:bg-gray-100"
+              >
+                <i class="fas fa-share-square fa-xl text-[color:var(--color-primary)]" />
+              </div>
+            </template>
+            <template #content>
+              <div class="flex justify-between items-center text-xl px-5 mb-3 h-16">
+                <a
+                  class="w-14"
+                  :href="stringFormat(socialSharingService.url_template, pageUrl)"
```
Comment (trueboroda) [inline client-app/shared/catalog/components/product-price-block.vue:47]: Maybe it be more readeble if move it a small code to method with name some like getSocialPostUrlForProduct()

**Review by trueboroda — APPROVED** [review]

File: client-app/shared/catalog/components/product-price-block.vue [hunk 3]
```diff
@@ -82,7 +119,15 @@ function addToList() {
   });
 }
 
+function getProductSocialShareUrl(urlTemplate: string, url: string): string {
```
Comment (Flanker72) [inline client-app/shared/catalog/components/product-price-block.vue:122]: Why not to use stringFormat directly in the template? We have a proxy method with the same arguments for now.

File: client-app/shared/catalog/components/product-price-block.vue [hunk 4]
```diff
@@ -30,34 +30,68 @@
...
+                    'text-[color:var(--color-primary)]': !shareProductPopoverShown,
+                    'text-gray-400': shareProductPopoverShown,
+                  }"
+                />
+              </div>
+            </template>
+            <template #content>
+              <div class="flex justify-between items-center px-5 mt-1.5 mb-7 space-x-6">
+                <a
+                  class=""
```
Comment (Flanker72) [inline client-app/shared/catalog/components/product-price-block.vue:56]: It's better to add target="_blank" to open share link in a separate tab.

**Review by artmakarov — APPROVED** [review]

**Pull request merged**